### PR TITLE
activities => working with avatars

### PR DESCRIPTION
**Pull request by avolchivschi**

### CODE REVIEW

File: src/assets/styles/_activities.scss [hunk 1]
```diff
@@ -19,49 +19,100 @@
...
+    right: 5px;
+    z-index: 1;
+    font-size: 40px;
+    cursor: pointer;
+    background: var(--yellowGradient);
+    -webkit-background-clip: text;
+    -webkit-text-fill-color: transparent;
+  }
+
+  .create-activity-icon:hover {
```
Comment (nlaptedulce) [inline src/assets/styles/_activities.scss]: you can write hover in '.create-activity-icon{ ... &:hover{....}}', and don't write     -webkit-background-clip: text;
    -webkit-text-fill-color: transparent; in hover, you have this in create-activity-icon

File: src/assets/styles/_profile.scss [hunk 2]
```diff
@@ -1,55 +1,59 @@
...
     font-family: 'Raleway', sans-serif;
     font-size: 14px;
     font-weight: 400;
-    color: #777;
+    color: var(--gray);
   }
-  .seniorityText{
-    text-shadow: 1px 1px #777;
+
+  .seniorityText {
```
Comment (nlaptedulce) [inline src/assets/styles/_profile.scss]: rename this class with kebab-case

File: src/assets/styles/_profile.scss [hunk 3]
```diff
@@ -70,55 +74,52 @@
     height: 200px;
     width: 200px;
     border-radius: 50%;
-    border: 5px solid white;
+    border: 5px solid var(--white);
   }
 
   .image-out-container .greenIcon {
```
Comment (nlaptedulce) [inline src/assets/styles/_profile.scss]: rename more sugestiv and with kebak-case

File: src/assets/styles/_profile.scss [hunk 4]
```diff
@@ -70,55 +74,52 @@
...
-    background-color: #4cd137;
+    background-color: var(--greenLight);
     position: absolute;
     right: 30px;
     bottom: 10px;
     height: 30px;
     width: 30px;
-    border: 5px solid white;
+    border: 5px solid var(--white);
```
Comment (nlaptedulce) [inline src/assets/styles/_profile.scss]: for white and black or basic colors you don't need variables

File: src/components/activity/ActivityList.vue [hunk 5]
```diff
@@ -149,130 +146,44 @@
...
+      return {
+        max: 10,
+        results: [],
+        pagination: {
+          currentPage: 1,
+          per_page: 9,
+          numPages: 1,
+          numResults: 1,
+        },
+      };
```
Comment (nlaptedulce) [inline src/components/activity/ActivityList.vue]: remove commas

File: src/components/activity/ActivityList.vue [hunk 6]
```diff
@@ -149,130 +146,44 @@
...
+    },
+    computed: {
+      ...mapState('account', ['user']),
```
Comment (nlaptedulce) [inline src/components/activity/ActivityList.vue]: remove comma

Comment (nlaptedulce) [reply src/components/activity/ActivityList.vue]: ### Component/instance options order

- name
- inject
- components
- props
- data
- created
- mounted
- computed
- methods
- destroyed

File: src/components/activity/ApplicantsList.vue [hunk 7]
```diff
@@ -9,47 +9,47 @@
...
+    },
+    methods: {
+      getListApplicants() {
+        ActivityService.getApplicantsList(this.$route.params.idActivity).then(
+          response => {
+            this.form.results = response.data.results;
+            this.isBusy = false;
+          },
+        );
+      },
```
Comment (nlaptedulce) [inline src/components/activity/ApplicantsList.vue]: commas

File: src/components/activity/ApplicantsList.vue [hunk 8]
```diff
@@ -9,47 +9,47 @@
...
+          results: [],
+        },
+      };
+    },
+    mounted() {
+      this.getListApplicants();
+    },
+    methods: {
+      getListApplicants() {
+        ActivityService.getApplicantsList(this.$route.params.idActivity).then(
```
Comment (nlaptedulce) [inline src/components/activity/ApplicantsList.vue:46]: then in new line

File: src/components/activity/CreateActivity.vue [hunk 9]
```diff
@@ -133,60 +133,70 @@
...
+          public: true,
+          technologies: [],
+          types: [],
+        },
+        timeStartApplication: '',
+        timeStartDeadline: '',
+      };
+    },
+    created() {
+      let startTime = new Date();
```
Comment (nlaptedulce) [inline src/components/activity/CreateActivity.vue:163]: one line after let

File: src/components/activity/CreateActivity.vue [hunk 10]
```diff
@@ -133,60 +133,70 @@
...
+      startTime.setDate(startTime.getDate() + 1);
+      this.form.application_deadline = moment(startTime).toISOString();
+      this.timeStartApplication = moment(startTime).toISOString();
+      this.setDeadline();
+    },
+    methods: {
+      setDeadline() {
+        let deadline = moment(this.form.application_deadline).add(1, 'days');
```
Comment (nlaptedulce) [inline src/components/activity/CreateActivity.vue:172]: same

File: src/components/activity/CreateActivity.vue [hunk 11]
```diff
@@ -133,60 +133,70 @@
...
+        this.form.final_deadline = moment(deadline).toISOString();
+        this.timeStartDeadline = moment(deadline).toISOString();
+      },
+      createActivity() {
+        let activity = JSON.parse(JSON.stringify(this.form));
```
Comment (nlaptedulce) [inline src/components/activity/CreateActivity.vue:178]: same

File: src/components/activity/EditActivity.vue [hunk 12]
```diff
@@ -159,75 +158,84 @@
...
+          status: 0,
+          public: true,
+          technologies: [{ id: null }],
+          types: [],
+        },
+        edited: false,
+        technologiesList: [],
+      };
+    },
+    computed: {
```
Comment (nlaptedulce) [inline src/components/activity/EditActivity.vue:188]: ### Component/instance options order

- name
- inject
- components
- props
- data
- created
- mounted
- computed
- methods
- destroyed

File: src/components/activity/Technologies.vue [hunk 13]
```diff
@@ -1,46 +1,43 @@
...
-      :multiple="true"
-      :taggable="true"
-      class="row col-md-12"
-      @input="updateTechnologies($event)"
-    />
-  </b-form-group>
+  <multiselect
+    v-model="technologies"
+    placeholder="Search a technology"
+    label="name" track-by="id"
```
Comment (nlaptedulce) [inline src/components/activity/Technologies.vue]: track-by write in new line

File: src/components/activity/Technologies.vue [hunk 14]
```diff
@@ -1,46 +1,43 @@
...
+    mounted() {
+      this.selectedTechnologies = this.technologies;
+      UserApi.getTechnologies().then((response) => {
+        this.formTechnologies = response.data;
+      });
+    },
+    methods: {
+      updateTechnologies(technologies) {
+        this.$emit('changeTechnologiesList', technologies);
+      },
```
Comment (nlaptedulce) [inline src/components/activity/Technologies.vue]: commas

File: src/components/activity/Types.vue [hunk 15]
```diff
@@ -1,47 +1,46 @@
...
-        @input="updateTypes($event)"
-      />
-    </b-form-group>
-  </div>
+  <b-form-group>
+    <label class="typo__label ml-3 text-color-activity">Types:</label>
+    <multiselect
+      v-model="types"
+      placeholder="Search a type"
+      label="name" track-by="id"
```
Comment (nlaptedulce) [inline src/components/activity/Types.vue]: same

File: src/components/activity/Types.vue [hunk 16]
```diff
@@ -1,47 +1,46 @@
...
     },
     methods: {
-      updateTypes (types) {
-        this.$emit('changeTypesList', types)
-      }
-    }
-  }
+      updateTypes(types) {
+        this.$emit('changeTypesList', types);
+      },
```
Comment (nlaptedulce) [inline src/components/activity/Types.vue]: commas

File: src/components/activity/UsersList.vue [hunk 17]
```diff
@@ -1,23 +1,34 @@
...
-          :taggable="true"
-        />
-      </b-form-group>
-    </b-form>
+  <div class="activity-list">
+    <div class="pt-2 mb-5 pb-5">
+      <section class="section-style">
+        <b-container>
+          <b-card class="mx-auto border-warning">
+
```
Comment (nlaptedulce) [inline src/components/activity/UsersList.vue]: dremove this line

File: src/components/activity/UsersList.vue [hunk 18]
```diff
@@ -27,8 +38,31 @@
...
+            this.$toast.open({
+              message: 'User invited with success!',
+              type: 'success',
+              position: 'top-right',
+              duration: 3000,
+              dismissible: true,
+            });
+          })
+          .catch((error) => {
+            let message = error.response.data.message;
```
Comment (nlaptedulce) [inline src/components/activity/UsersList.vue:55]: after let new line

File: src/components/auth/Registration.vue [hunk 19]
```diff
@@ -185,9 +183,10 @@
             });
             this.$router.push('/login');
           })
-          .catch(() => {
+          .catch((error) => {
+            let message = error.response.data.message;
```
Comment (nlaptedulce) [inline src/components/auth/Registration.vue:188]: same

File: src/components/user/EditProfile.vue [hunk 20]
```diff
@@ -0,0 +1,240 @@
...
+        <b-form-input
+          id="firstName"
+          v-model="form.name"
+          v-validate.continues="'required|min:3|max:20'"
+          name="name"
+          type="text"
+          :class="{ 'is-invalid': errors.has('name') }"
+        />
+
+        <span v-if="errors.has('name')" class="invalid-feedback">{{
```
Comment (nlaptedulce) [inline src/components/user/EditProfile.vue]: it's not good

File: src/components/user/EditProfile.vue [hunk 21]
```diff
@@ -0,0 +1,240 @@
...
+        <b-form-input
+          id="lastName"
+          v-model="form.surname"
+          v-validate.continues="'required|min:3|max:20'"
+          name="surname"
+          type="text"
+          :class="{ 'is-invalid': errors.has('surname') }"
+        />
+
+        <span v-if="errors.has('surname')" class="invalid-feedback">{{
```
Comment (nlaptedulce) [inline src/components/user/EditProfile.vue]: same

File: src/components/user/EditProfile.vue [hunk 22]
```diff
@@ -0,0 +1,240 @@
...
+      form: {
+        position: '',
+        seniority: null,
+        name: '',
+        surname: '',
+        biography: '',
+        location: '',
+        technologies: [
+          {
+            id: null,
```
Comment (nlaptedulce) [inline src/components/user/EditProfile.vue]: it will be better { id:null}, in one line

File: src/components/user/EditProfile.vue [hunk 23]
```diff
@@ -0,0 +1,240 @@
...
+        { value: 'GUA', text: 'GUA' },
+        { value: 'LYO', text: 'LYO' },
+      ],
+      options3: [
+        { value: 'DEV', text: 'DEV' },
+        { value: 'PO', text: 'PO' },
+        { value: 'QA', text: 'QA' },
+      ],
+    }),
+    computed: {
```
Comment (nlaptedulce) [inline src/components/user/EditProfile.vue:188]: order

File: src/components/user/EditProfile.vue [hunk 24]
```diff
@@ -0,0 +1,240 @@
...
+    mounted() {
+      UserApi.userInfo(this.userId)
+        .then(response => {
+          this.form = response.data;
+          if (this.form.avatar) {
+            this.previewImage = basePath + '/' + this.form.avatar.original;
+            this.form.avatar = null;
+          }
+        });
+      UserApi.getTechnologies(this.userId)
```
Comment (nlaptedulce) [inline src/components/user/EditProfile.vue:202]: new line before

File: src/components/user/EditProfile.vue [hunk 25]
```diff
@@ -0,0 +1,240 @@
...
+          this.form.avatar = this.previewImage;
+          console.log(this.previewImage);
+        };
+      },
+      edit() {
+          const data = {
+            ...this.form,
+            id: this.userId,
+          };
+          console.log(this.form);
```
Comment (nlaptedulce) [inline src/components/user/EditProfile.vue]: remove console

File: src/components/user/EditProfile.vue [hunk 26]
```diff
@@ -0,0 +1,240 @@
...
+              this.$toast.open({
+                message: 'Please complete all required fields',
+                type: 'error',
+                position: 'top-right',
+                duration: 3000,
+                dismissible: true,
+              });
+            });
+      }
+    },
```
Comment (nlaptedulce) [inline src/components/user/EditProfile.vue]: commas

File: src/views/TheHeader.vue [hunk 27]
```diff
@@ -89,10 +103,10 @@
...
           duration: 3000,
-          dismissible: true
+          dismissible: true,
         });
-        this.$router.push('/login')
-      }
-    }
-  }
+        this.$router.push('/login');
+      },
```
Comment (nlaptedulce) [inline src/views/TheHeader.vue]: same

File: src/assets/styles/_activities.scss (hunk 1 above)
Comment (avolchivschi) [reply src/assets/styles/_activities.scss]: If I delete webkit in one or another part it didn't work..

**Pull request merged**